### PR TITLE
Feat/find one

### PR DESCRIPTION
**Pull request by bas080**

### CODE REVIEW

File: src/repository.js [hunk 1]
```diff
@@ -89,6 +90,33 @@ export class Repository {
...
+   * Performs a find. It tries to return a promise that resolves to an entity.
+   * Not a list of entities.
+   *
+   * @param {{}|number|string} criteria Criteria to add to the query. A plain string or number will be used as relative path.
+   * @param {boolean}          [raw]    Set to true to get a POJO in stead of populated entities.
+   *
+   * @return {Promise<Entity>}
+   */
+  findOne(...args) {
+    return this.find(...args).then(records => {
```
Comment (RWOverdijk) [inline src/repository.js]: Add a limit 1 to the criteria.

File: src/repository.js [hunk 2]
```diff
@@ -89,6 +90,33 @@ export class Repository {
...
+
+      if (records.length > 1) {
+        logger.warn(`${this.getResource}.findOne returned ${records.length} items`);
+      }
+
+      if (records.length > 0) {
+        return records[0];
+      }
+
+      return Promise.reject();
```
Comment (RWOverdijk) [inline src/repository.js]: Why a reject?

File: test/repository.spec.js [hunk 3]
```diff
@@ -92,6 +92,16 @@ describe('Repository', function() {
     });
   });
 
+  describe('.findOne', function() {
+    it('performs a find and resolves to a single item', function() {
+      let repository = constructRepository('find-one-test');
+
+      repository.findOne().then(response => {
+        expect(response.method).toEqual('GET')
```
Comment (RWOverdijk) [inline test/repository.spec.js:100]: This doesn't really test the method.

**Review by RWOverdijk — CHANGES_REQUESTED** [review]

File: src/repository.js (hunk 1 above)
Comment (VMBindraban) [inline src/repository.js]: Add `limit: 1` to criteria. You can override this. (since it's a findOne)

Comment (bas080) [reply src/repository.js]: Should it then no longer contain the length checks?

EDIT: should it be the responsibility of the server instead?

File: src/repository.js (hunk 2 above)
Comment (bas080) [reply src/repository.js]: If it fails to find**One** it should reject.

File: test/repository.spec.js (hunk 3 above)
Comment (bas080) [reply test/repository.spec.js:100]: I could mock the find method. That way I can test the logic in the `then`.

File: src/repository.js (hunk 2 above)
Comment (RWOverdijk) [reply src/repository.js]: https://github.com/bas080/aurelia-orm/blob/193be843c02b423574b9e626279e7ffd6a3d2278/src/repository.js#L143

**Review by RWOverdijk — CHANGES_REQUESTED** [review]

**Comment by bas080** [comment]
@RWOverdijk, added requested changes.

**Pull request closed without merging**